src/GameServer/Systems/Inventory/InventoryManager.cs

```diff
@@ -142,7 +142,8 @@ private int GetVirtualItemValue(int itemId)
             if (updatedItem != null)
             {
                 // Add a reference by Guid
-                GuidMap.Add(updatedItem.Guid, updatedItem);
+                if(!GuidMap.TryGetValue(updatedItem.Guid, out _))
```

Comment:
This check can be avoided entirely by the built-in "insert if not present" feature of c# dictionaries.
```suggestion
                GuidMap[updatedItem.Guid] = updatedItem
```